lib/Conversion/StableHLOToTTIR/StableHLOToTTIRPass.cpp

```diff
@@ -37,12 +39,28 @@ class StablehloTypeConverter : public TypeConverter {
              "only ranked tensor type supported");
       return type;
     });
+
+    // TTNN doesn't support either scalars or boolean data. This transformation
+    // converts boolean to bfloat16 and scalars to 1-D tensors.
     addConversion([&](RankedTensorType type) -> RankedTensorType {
-      if (type.getShape().size() == 0) {
-        auto ElementType = type.getElementType();
-        return RankedTensorType::get({1}, ElementType);
+      bool changed = false;
+      Type elementType = type.getElementType();
+      llvm::ArrayRef<int64_t> shape = type.getShape();
+      // Convert the element type to bfloat16 if the input is boolean.
+      if (type.getElementTypeBitWidth() == 1) {
+        elementType = BFloat16Type::get(elementType.getContext());
+        changed = true;
       }
-      return type;
+
+      // Return 1-D tensor instead of scalar with either original element type
+      // or bfloat16 (in case of boolean input).
+      if (shape.size() == 0) {
+        return RankedTensorType::get({1}, elementType);
```

Comment:
To avoid multiple returns, can you simply it like this:

Type elementType = type.getElementType();
llvm::ArrayRef<int64_t> shape = type.getShape();
if (type.getElementTypeBitWidth() == 1) {
  elementType = BFloat16Type::get(elementType.getContext());
}
if (shape.size() == 0) {
   shape = llvm::ArrayRef<int64_t>({1});
}

return RankedTensorType::get(shape, elementType);

Reply:
This was my initial implementation but it doesn't work as we had discussed earlier. I can change it as below

```
bool changed = type.getShape().size() == 0;
std::vector<int64_t> shapeVec({1});
llvm::ArrayRef<int64_t> shape = changed ? llvm::ArrayRef<int64_t>(shapeVec) : type.getShape();

if (type.getElementTypeBitWidth() == 1) {
   elementType = BFloat16Type::get(elementType.getContext());
   changed = true;
}
return changed ? RankedTensorType::get(shape, elementType) : type;
```

Reply:
updated as suggested.